platform/commonUI/dialog/src/DialogService.js

```diff
@@ -157,6 +150,99 @@ define(
             );
         };
 
+        /**
+         * Tests if a dialog can be displayed. A modal dialog may only be
+         * displayed if one is not already visible.
+         * Will log a warning message if it can't display a dialog.
+         * @returns {boolean} true if dialog is currently visible, false
+         * otherwise
+         */
+        DialogService.prototype.canShowDialog = function(dialogModel){
+            if (this.dialogVisible){
+                // Only one dialog should be shown at a time.
+                // The application design should be such that
+                // we never even try to do this.
+                this.$log.warn([
+                    "Dialog already showing; ",
+                    "unable to show ",
+                    dialogModel.title
+                ].join(""));
+
+                return false;
+            } else {
+                return true;
+            }
+        };
+
+        /**
+         * A user action that can be performed from a blocking dialog. These
+         * actions will be rendered as buttons within a blocking dialog.
+         *
+         * @typedef DialogAction
```

Comment:
Even clearer. Or `DialogOption`, so as not to presume presentation as a button.

We could also just consider passing in an `Action` here - there's nothing in the interface that assumes an `Action` effects a specific domain object, and all the metadata is appropriate for presentation as a button.

(At this point, I'm just overthinking it.)